pkg/virt/snapshot.go

```diff
@@ -129,20 +129,21 @@ func (vm *VM) ListMatchingSnapshots(regexes []string) ([]Snapshot, error) {
 // FreeSnapshots is a function that takes a slice of snapshots and frees any
 // associated libvirt.DomainSnapshot. Usually, this is called after
 // ListMatchingSnapshots with a "defer" statement.
-func FreeSnapshots(snapshots []Snapshot) {
+func FreeSnapshots(log *zap.SugaredLogger, snapshots []Snapshot) {
```

Comment:
Code in "pkg" is meant to be importable by others. What if a potential user does not want to use Zap? A parameter value of "nil" would produce errors, since we do not check for. What is your opinion on this? Should we check in every method whether log is nil or go without it and make the user to initialize a Zap Logger may not want?

Reply:
We could create an interface `Logger` that needs to implement the functions we need: `Info`, `Infof`, `Error`, `Errorf`, ... 
This means a bit more code but more flexibility in regards to which logger is used.

Reply:
Using an interface now.